actors/states/election.go

```diff
@@ -0,0 +1,55 @@
+package states
+
+import (
+	addr "github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+
+	"github.com/filecoin-project/specs-actors/v2/actors/builtin/miner"
+	"github.com/filecoin-project/specs-actors/v2/actors/builtin/power"
+	"github.com/filecoin-project/specs-actors/v2/actors/builtin/reward"
+	"github.com/filecoin-project/specs-actors/v2/actors/util/adt"
+)
+
+// Checks for miner election eligibility.
+// A miner must satisfy conditions on both the immediate parent state, as well as state at the
+// Winning PoSt election lookback.
+
+// Tests whether a miner is eligible to win an election given the immediately prior state of the
+// miner and system actors.
+func MinerEligibleForElection(store adt.Store, mstate *miner.State, pstate *power.State, rstate *reward.State, maddr addr.Address, currEpoch abi.ChainEpoch) (bool, error) {
+	// Non-empty power claim.
+	if claim, found, err := pstate.GetClaim(store, maddr); err != nil {
+		return false, err
+	} else if !found {
+		return false, err
+	} else if claim.QualityAdjPower.LessThanEqual(big.Zero()) {
+		return false, err
+	}
+
+	// No fee debt.
+	if !mstate.IsDebtFree() {
+		return false, nil
+	}
+
+	// No active consensus faults.
+	if mInfo, err := mstate.GetInfo(store); err != nil {
+		return false, err
+	} else if miner.ConsensusFaultActive(mInfo, currEpoch) {
+		return false, nil
+	}
+
+	// IP requirement is sufficient to cover fee for a consensus fault
+	electionRequirement := miner.ConsensusFaultPenalty(rstate.ThisEpochReward)
+	if mstate.InitialPledge.LessThan(electionRequirement) {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// Tests whether a miner is eligible for election given a Winning PoSt lookback state.
+func MinerEligibleForElectionAtPoStLookback(store adt.Store, pstate *power.State, mAddr addr.Address) (bool, error) {
```

Comment:
Nit. This function returning true doesn't imply the miner was eligible at PoSt lookback. Also this method doesn't deal with lookback directly. `MinerMeetsConsensusMinimumAtPoStLookback`would be more accurate. `MinerMeetsPoStLookbackEligibilityRequirement`? I'm not convinced this function adds much over calling `pstate.MinerNominalPowerMeetsConsensusMinimum(store, mAddr)` directly. 

The comment kind of gets at it, but it would be better to to directly call out "pstate must be the prior state of the power actor at the PoSt lookback epoch". This is an unusual thing in the actor code.

Reply:
What this adds is that when the eligibility requirements change, we can encapsulate that change here (or at least force downstream ack of it through a type signature change). Such a change is likely to be accompanied by other actor changes so we can cover the whole thing and reduce coordination with Lotus.

Thanks for the name suggestion.